sql: remove usage of OptionalNodeIDErr in createPlanForInterleavedJoin

asubiotto opened the pull request:
Depends on #52562 

The goal was to get the gateway node ID, which is already stored in the
physical planner as a SQLInstanceID. This was erroneously resulting in an
UnsupportedWithMultiTenancy error when attempting to execute an interleaved
join as a tenant.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52564)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
2nd commit <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52564#-:-MENwqO287dtn3HswXQj:b-5panh4)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Commit:
sql: remove usage of OptionalNodeIDErr in createPlanForInterleavedJoin

The goal was to get the gateway node ID, which is already stored in the
physical planner as a SQLInstanceID. This was erroneously resulting in an
UnsupportedWithMultiTenancy error when attempting to execute an interleaved
join as a tenant.

Release note: None

asubiotto commented:
bors r=yuzefovich
